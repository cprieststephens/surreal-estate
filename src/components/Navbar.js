import React from "react";
import { Link } from "react-router-dom";
import logo from "../img/logo.png";
import "../styles/navbar.css";

const Navbar = ({ onSetUserID }) => {
  const handleLogin = () => {};
  const handleLogout = () => {};

  return (
    <div className="navbar">
      <img className="navbar-logo" src={logo} alt="Surreal Estate logo" />
      <h2 className="navbar-title">Surreal Estate</h2>
      <div className="navbar-links">
        <Link className="navbar-links-item" to="/">
          View Properties
        </Link>
        <Link className="navbar-links-item" to="add-property">
          Add a Property
        </Link>
        <Link className="navbar-links-item" to="sign-up">
          Sign Up
        </Link>
      </div>
      {onSetUserID ? (
        <button className="navbar-button" type="button" onClick={handleLogout}>
          Sign Out
        </button>
      ) : (
        <button className="navbar-button" type="button" onClick={handleLogin}>
          Sign In
        </button>
      )}
    </div>
  );
};

export default Navbar;
