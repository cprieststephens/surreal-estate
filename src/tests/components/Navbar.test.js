import React from "react";
import { MemoryRouter } from "react-router-dom"
import { render, screen } from "@testing-library/react";
import Navbar from "../../components/Navbar";

describe("Navbar", () => {
  const validProps = {
    setUserID: jest.fn(),
    userID: "11111",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navbar setUserID={validProps.setUserID} userID={validProps.userID} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct logo", () => {
    render(
      <MemoryRouter>
        <Navbar setUserID={validProps.setUserID} userID={validProps.userID} />
      </MemoryRouter>
    );
    const logo = screen.getByAltText("Surreal Estate logo");

    expect(logo).toBeInTheDocument();
  });

  it("renders the correct title", () => {
    render(
      <MemoryRouter>
        <Navbar setUserID={validProps.setUserID} userID={validProps.userID} />
      </MemoryRouter>
    );
    const surrealEstate = screen.getByText(/surreal estate/i);

    expect(surrealEstate).toBeInTheDocument();
  });

  it("renders a button", () => {
    render(
      <MemoryRouter>
        <Navbar setUserID={validProps.setUserID} userID={validProps.userID} />
      </MemoryRouter>
    );
    const signOutButton = screen.getByRole("button");

    expect(signOutButton).toBeInTheDocument();
  });
});
