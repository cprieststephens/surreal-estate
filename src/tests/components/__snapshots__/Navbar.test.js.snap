// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="Surreal Estate logo"
      class="navbar-logo"
      src="logo.png"
    />
    <h2
      class="navbar-title"
    >
      Surreal Estate
    </h2>
    <div
      class="navbar-container"
    >
      <ul
        class="navbar-links"
      >
        <li>
          <a
            class="navbar-links-item"
            href="/"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-house navbar-icon"
              data-icon="house"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
                fill="currentColor"
              />
            </svg>
            <span
              class="navbar-text"
            >
              View Properties
            </span>
          </a>
        </li>
        <li>
          <a
            class="navbar-links-item"
            href="/add-property"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-plus navbar-icon"
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                fill="currentColor"
              />
            </svg>
            <span
              class="navbar-text"
            >
              Add a Property
            </span>
          </a>
        </li>
        <li>
          <a
            class="navbar-links-item"
            href="/saved-properties"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-star navbar-icon"
              data-icon="star"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
                fill="currentColor"
              />
            </svg>
            <span
              class="navbar-text"
            >
              Saved Properties
            </span>
          </a>
        </li>
      </ul>
      <button
        class="navbar-button"
        type="button"
      >
        Sign Out
      </button>
    </div>
    <div
      class="navbar__alert"
    />
  </div>
</DocumentFragment>
`;
